### PR TITLE
Stop building for .NET 6

### PR DESCRIPTION
**Pull request by RussKie**
Build the shippable artifacts with net462,netcoreapp3.1,net8.0, and run tests for net462,netcoreapp3.1,net6.0,net7.0,net8.0 (#4036)

Resolves dotnet/r9#423
Companion to dotnet/extensions#4036

### CODE REVIEW

File: eng/Packages/TestOnly-net7.0.props [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup Condition="'$(TargetFramework)' == 'net7.0'">
+    <PackageVersion Include="Microsoft.AspNetCore.Mvc.Testing" Version="7.0.0" />
```
Comment (joperezr) [inline eng/Packages/TestOnly-net7.0.props:4]: I'm not sure I'm following why this is needed now.

Comment (joperezr) [reply eng/Packages/TestOnly-net7.0.props:4]: Oh we are adding 7.0 for tests, got it, disregard.

**Review by joperezr — APPROVED** [review]

**Review by geeknoid — APPROVED** [review]

**Comment by geeknoid** [comment]
Note that in dotnet/extensions, I've just simplified the stuff in eng/Packages.

**Comment by RussKie** [comment]
> Note that in dotnet/extensions, I've just simplified the stuff in eng/Packages.

👍  I'll copy those in a follow up pull request.

**Pull request merged**